flume-hdfs-sink/src/main/java/com/wipro/ats/bdre/flume/sink/hdfs/SequenceFileSerializerFactory.java

```diff
@@ -25,27 +25,27 @@
 
 public class SequenceFileSerializerFactory {
 
-  private static final Logger logger =
+  private static final Logger LOGGER =
       LoggerFactory.getLogger(SequenceFileSerializerFactory.class);
 
+  private SequenceFileSerializerFactory(){}
+
   /**
    * {@link Context} prefix
    */
   static final String CTX_PREFIX = "writeFormat.";
 
-  @SuppressWarnings("unchecked")
+  @SuppressWarnings("squid:S1166")
   static SequenceFileSerializer getSerializer(String formatType,
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) The Cyclomatic Complexity of this method "getSerializer" is 14 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)